homeassistant/components/evohome/climate.py

```diff
@@ -65,8 +62,11 @@
 HA_PRESET_TO_EVO = {v: k for k, v in EVO_PRESET_TO_HA.items()}
 
 
-def setup_platform(
-    hass: HomeAssistantType, hass_config: ConfigType, add_entities, discovery_info=None
+async def async_setup_platform(
+    hass: HomeAssistantType,
+    hass_config: ConfigType,
+    async_add_entities,
+    discovery_info=None,
 ) -> None:
     """Create the evohome Controller, and its Zones, if any."""
     broker = hass.data[DOMAIN]["broker"]
```

Comment:
We normally add a guard clause here that checks if discovery_info is None and return is so, when we load platforms via discovery.